ledger/participant-integration-api/src/main/resources/db/migration/oracle/V1__Init.sql

```diff
@@ -81,9 +81,8 @@ CREATE TABLE packages
 -- Table for storing a log of ledger configuration changes and rejections.
 CREATE TABLE configuration_entries
 (
-    ledger_offset    VARCHAR2(4000)  not null,
-    recorded_at      timestamp       not null, -- with time zone
-
+    ledger_offset    VARCHAR2(4000)  not null primary key,
+    recorded_at      timestamp       not null, -- TODO BH: in postgres this is with timezone
```

Comment:
The Postgres schema uses `timestamp`, which is [without](https://www.postgresql.org/docs/9.1/datatype-datetime.html) a timezone.

Reply:
every oracle timezone variant actually does store some time offset -- it looks like `TIMESTAMP WITH LOCAL TIME ZONE` is what the documentation recommends if you don't care about the time component, but i have experienced this get screwy in the past when you don't explicitly run your application servers and DB as UTC, as it shifts to local (particularly problematic for a global application across asia / emea / americas where a local shift actually changes the effective day)

we should probably add some tests at timezone shift boundaries (early morning hours asia where emea and americas are a day behind, late day americas where asia is a day ahead)

https://docs.oracle.com/en/database/oracle/oracle-database/19/nlspg/datetime-data-types-and-time-zone-support.html#GUID-2DFBD56C-5832-4AE8-831E-7EF1EFB7F0BB

```The TIMESTAMP WITH LOCAL TIME ZONE datatype stores the timestamp without time zone information. It normalizes the data to the database time zone every time the data is sent to and from a client. It requires 11 bytes of storage.```

```If you specify a date value without a time component, then the default time is midnight. If you specify a date value without a date, then the default date is the first day of the current month.

Oracle Database DATE columns always contain fields for both date and time. If your queries use a date format without a time portion, then you must ensure that the time fields in the DATE column are set to midnight. You can use the TRUNC (date) SQL function to ensure that the time fields are set to midnight, or you can make the query a test of greater than or less than (<, <=, >=, or >) instead of equality or inequality (= or !=). Otherwise, Oracle Database may not return the query results you expect.```

Reply:
changing to `TIMESTAMP WITH LOCAL TIME ZONE` then confuses conversion with handling the java.sql.timestampz field

so i think in short, let's stick with `TIMESTAMP` and i'll remove the todos

```14:42:48.465 [daml.index.db.threadpool.connection.jdbcledgerdaooracleappendonlyspec-1] ERROR c.d.p.s.appendonlydao.DbDispatcher - Exception while executing SQL query. Rolled back. , context: {metric=get_completions}
anorm.AnormException: TypeDoesNotMatch(Cannot convert oracle.sql.TIMESTAMPLTZ@18fd069f: oracle.sql.TIMESTAMPLTZ to Date for column ColumnName(.RECORD_TIME,Some(RECORD_TIME)))
- should serialize a valid fetch *** FAILED *** (21 milliseconds)
  anorm.AnormException: TypeDoesNotMatch(Cannot convert oracle.sql.TIMESTAMPLTZ@18fd069f: oracle.sql.TIMESTAMPLTZ to Date for column ColumnName(.RECORD_TIME,Some(RECORD_TIME)))```

Reply:
just a note: running DB servers at UTC is pretty much a must for postgres as well